src/main/java/org/truffleruby/core/time/RubyDateFormatter.java

```diff
@@ -655,18 +655,21 @@ public static Rope formatToRopeBuilderFast(Token[] compiledPattern, ZonedDateTim
 
                 case FORMAT_NANOSEC: {
                     final int nano = dt.getNano();
-                    final LazyIntRope nanoRope = new LazyIntRope(nano);
+                    final LazyIntRope microSecondRope = new LazyIntRope(nano / 1000);
 
                     final int length = 6;
-                    final int padding = length - nanoRope.characterLength();
+                    final int padding = length - microSecondRope.characterLength();
 
                     if (padding == 0) {
-                        appendRope = nanoRope;
+                        appendRope = microSecondRope;
                     } else if (padding < 0) {
```

Comment:
Is this case even possible actually?
I would think not, because the maximum value is 999_999_999 nanos, / 100 so 999_999 and that can never be bigger than 6 digits.

Reply:
I agree: `padding = 6 - (new LazyIntRope(nano / 1000)).characterLength()`, and `nano` is a 9-digit integer at most, so the value of `(new LazyIntRope(nano / 1000)).characterLength()` can never be larger than the 6 it’s being subtracted from.

Reply:
You're right. Prior to adding the division in, we needed to account for fractional microseconds. As far as I can tell, those always just get truncated, which is why the division is safe. I forgot to clean this branch up afterwards.

Reply:
Update pushed, along with a couple of comments so anyone coming across it will better understand the various width guarantees.